src/ORM/Marshaller.php

```diff
@@ -439,6 +443,51 @@ protected function _loadBelongsToMany($assoc, $ids)
     }
 
     /**
+     * Call translation merge. Validations errors during merge will be added to `$errors` param
+     *
+     * @param \Cake\Datasource\EntityInterface $original The original entity
+     * @param array $data key value list of languages with fields to be merged into the translate entity
+     * @param array $errors array with entity errors
+     * @param array $options list of options
+     * @return array|null
+     */
+    protected function _mergeTranslations($original, array $data, array &$errors, array $options = [])
+    {
+        $result = $this->_table->mergeTranslations($original, $data, $this, $options);
+
+        if (is_array($result)) {
+            if ((bool)$result[1]) {
+                $errors['_translations'] = $result[1];
+            }
+            $result = $result[0];
+        }
+
+        return $result;
+    }
+
+    /**
+     * Return if table contains translate behavior or we specificate to use via `translations` options.
+     *
+     * In case that $options has `fieldList` option and `_translations` field is not present inside it, it will include
+     *
+     * ### Options:
+     *
+     * - translations: Set to false to disable translations
+     *
+     * @param array $options List of options
+     * @return bool
+     */
+    protected function _hasTranslations(array &$options = [])
```

Comment:
Please don't use pass by reference to sneak multiple return values in. Instead eithe have separate methods, or return a tuple that can be unpacked with list()